File: Easy/plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        str_d = [str(d) for d in digits]
        l_int = int("".join(str_d))+1
        output = []

        for i in range(len(str(l_int))):
            output.append(int(str(l_int)[i]))
            
        return output

File: Easy/Swift/plus_one.swift
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var dummy = digits
        dummy[dummy.endIndex-1] += 1

        var output = [Int]()
        
        var digitMem = 0
        for d in dummy.reversed() {
            let nd = d+digitMem
            digitMem = 0

            if nd == 10 {
                digitMem+=1
                output.append(0)
            } else {
                output.append(nd)
            }
        }
        if digitMem == 1 { output.append(digitMem) }
        
        return output.reversed()
    }
}
